core/iwasm/aot/aot_loader.c

```diff
@@ -1026,8 +1030,21 @@ load_mem_init_data_list(const uint8 **p_buf, const uint8 *buf_end,
         data_list[i]->offset.init_expr_type = init_value.init_expr_type;
         data_list[i]->offset.u = init_value.u;
         data_list[i]->byte_count = byte_count;
-        read_byte_array(buf, buf_end, data_list[i]->bytes,
-                        data_list[i]->byte_count);
+        data_list[i]->bytes = NULL;
+        if (module->is_binary_freeable) {
```

Comment:
There is similar handling in wasm loader, but it clones memory init data only when both `is_load_from_file_buf` and `module->is_binary_freeable` are true:
https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/core/iwasm/interpreter/wasm_loader.c#L5791
note `is_load_from_file_buf` is passed to this fuction's caller `load_init_data_section`.

Do you think here we had better handle `data_lists[i]->bytes` like wasm_loader.c?

Reply:
Yes, it's better to handle them by same way

Reply:
But, pass the `is_load_from_file_buf ` will change many places, I think `module->is_binary_freeable` is enough for many use case